llama_index/vector_stores/tair.py

```diff
@@ -140,8 +150,8 @@ def add(self, embedding_results: List[NodeWithEmbedding]) -> List[str]:
                 "doc_id": result.ref_doc_id,
                 "text": result.node.get_text(),
             }
-            additional_metadata = node_to_metadata_dict(result.node)
-            attributes.update(additional_metadata)
+            if result.node.node_info is not None:
+                attributes.update(result.node.node_info)
```

Comment:
nit: could we use the utility in `llama_index/vector_stores/utils.py` to get metadata dict from the Node (and re-creating Node from metadata dict)?

Reply:
You can take a look at `llama_index/vector_stores/chroma.py` for reference.

Reply:
> nit: could we use the utility in `llama_index/vector_stores/utils.py` to get metadata dict from the Node (and re-creating Node from metadata dict)?

node_to_metadata_dict() put all the extra key/value pairs in Node.node_info into a single json string. This makes it impossible for Tair to filter documents by these key/value pairs. Is that ok?

Reply:
Yes, this is actually intended. we use `extra_info` as the fields for metadata filtering, and `node_info` for other misc data of arbitrary types.